src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -177,7 +177,7 @@ internal static BoundBlock AppendImplicitReturn(BoundBlock body, MethodSymbol me
                 ? (BoundStatement)BoundYieldBreakStatement.Synthesized(syntax)
                 : BoundReturnStatement.Synthesized(syntax, RefKind.None, null);
 
-            return body.Update(body.Locals, body.LocalFunctions, body.Statements.Add(ret));
+            return body.Update(body.Locals, body.LocalFunctions, body.HasUnsafeModifier, body.Statements.Add(ret));
```

Comment:
not part of your PR, but it seems so strange to me that in one case we preserve the `body.LocalFunctions` and in another case we replace it with `ImmutableArray<LocalFunctionSymbol>.Empty`.